Essentials/src/com/earth2me/essentials/services/ReserveEssentials.java

```diff
@@ -0,0 +1,1875 @@
+     * @return True if an account exists for this player, else false.
+     */
+    @Override
+    public CompletableFuture<Boolean> asyncHasAccount(String identifier) {
+        return CompletableFuture.supplyAsync(new Supplier<Boolean>() {
+
+            /**
+             * Gets a result.
+             *
+             * @return a result
+             */
+            @Override
+            public Boolean get() {
+                return Economy.playerExists(identifier);
+            }
+        });
+    }
+
+    /**
+     * Checks to see if an account exists for this identifier. This method should be used for player accounts.
+     * @param identifier The {@link UUID} of the account.
+     * @return True if an account exists for this player, else false.
+     */
+    @Override
+    public CompletableFuture<Boolean> asyncHasAccount(UUID identifier) {
+        return CompletableFuture.supplyAsync(new Supplier<Boolean>() {
+
+            /**
+             * Gets a result.
+             *
+             * @return a result
+             */
+            @Override
+            public Boolean get() {
+                return Economy.playerExists(getName(identifier));
+            }
+        });
+    }
+
+    /**
+     * Attempts to create an account for this identifier. This method should be used for non-player accounts.
+     * @param identifier The identifier of the account.
+     * @return True if an account was created, else false.
+     */
+    @Override
+    public boolean createAccount(String identifier) {
+        if (hasAccount(identifier)) return false;
+        return Economy.createNPC(identifier);
+    }
+
+    /**
+     * Attempts to create an account for this identifier. This method should be used for player accounts.
+     * @param identifier The {@link UUID} of the account.
+     * @return True if an account was created, else false.
+     */
+    @Override
+    public boolean createAccount(UUID identifier) {
+        if (hasAccount(identifier)) return false;
+        return Economy.createNPC(getName(identifier));
+    }
+
+    /**
+     * Attempts to create an account for this identifier. This method should be used for non-player accounts.
+     * @param identifier The identifier of the account.
+     * @return True if an account was created, else false.
+     */
+    @Override
+    public CompletableFuture<Boolean> asyncCreateAccount(String identifier) {
+        return CompletableFuture.supplyAsync(() -> {
+            if (hasAccount(identifier)) return false;
+            return Economy.createNPC(identifier);
+        });
+    }
+
+    /**
+     * Attempts to create an account for this identifier. This method should be used for player accounts.
+     * @param identifier The {@link UUID} of the account.
+     * @return True if an account was created, else false.
+     */
+    @Override
+    public CompletableFuture<Boolean> asyncCreateAccount(UUID identifier) {
+        return CompletableFuture.supplyAsync(() -> {
```

Comment:
These async versions should delegate to the non-async versions (or the other way around) to reduce code duplication.